README.md

```diff
@@ -72,68 +89,153 @@ dependencies {
+If the flag management system you're using supports targeting, you can provide the input data using the [evaluation context](https://openfeature.dev/docs/reference/concepts/evaluation-context).
+
+```java
+// set a value to the global context
 OpenFeatureAPI api = OpenFeatureAPI.getInstance();
-Map<String, Value> attributes = new HashMap<>();
-attributes.put("appVersion", new Value(System.getEnv("APP_VERSION")));
-EvaluationContext apiCtx = new ImmutableContext(attributes);
+Map<String, Value> apiAttrs = new HashMap<>();
+apiAttrs.put("region", new Value(System.getEnv("us-east-1")));
+EvaluationContext apiCtx = new ImmutableContext(apiAttrs);
 api.setEvaluationContext(apiCtx);
 
-// request context
-Map<String, Value> attributes = new HashMap<>();
-attributes.put("email", new Value(session.getAttribute("email")));
-attributes.put("product", new Value(productId));
+// set a value to the client context
+Map<String, Value> clientAttrs = new HashMap<>();
+clientAttrs.put("region", new Value(System.getEnv("us-east-1")));
+EvaluationContext clientCtx = new ImmutableContext(clientAttrs);
+Client client = api.getInstance().getClient();
+client.setEvaluationContext(clientCtx);
+
+// set a value to the invocation context
+Map<String, Value> requestAttrs = new HashMap<>();
+requestAttrs.put("email", new Value(session.getAttribute("email")));
+requestAttrs.put("product", new Value("productId"));
 String targetingKey = session.getId();
-EvaluationContext reqCtx = new ImmutableContext(targetingKey, attributes);
+EvaluationContext reqCtx = new ImmutableContext(targetingKey, requestAttrs);
 
-// use merged contextual data to determine a flag value
 boolean flagValue = client.getBooleanValue("some-flag", false, reqCtx);
 ```
 
-### Events
+### Hooks
+
+[Hooks](https://openfeature.dev/docs/reference/concepts/hooks) allow for custom logic to be added at well-defined points of the flag evaluation life-cycle
+Look [here](https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Hook&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=Java) for a complete list of available hooks.
+If the hook you're looking for hasn't been created yet, see the [develop a hook](#develop-a-hook) section to learn how to build it yourself.
+
+Once you've added a hook as a dependency, it can be registered at the global, client, or flag invocation level.
+
+```java
+    // add a hook globally, to run on all evaluations
+    OpenFeatureAPI api = OpenFeatureAPI.getInstance();
+    api.addHooks(new ExampleHook());
+    
+    // add a hook on this client, to run on all evaluations made by this client
+    Client client = api.getClient();        
+    client.addHooks(new ExampleHook());
+    
+    // add a hook for this evaluation only
+    Boolean retval = client.getBooleanValue(flagKey, false, null,
+            FlagEvaluationOptions.builder().hook(new ExampleHook()).build());
```

Comment:
To align with all other examples, 

```suggestion
```java
  // add a hook globally, to run on all evaluations
  OpenFeatureAPI api = OpenFeatureAPI.getInstance();
  api.addHooks(new ExampleHook());
  
  // add a hook on this client, to run on all evaluations made by this client
  Client client = api.getClient();        
  client.addHooks(new ExampleHook());
  
  // add a hook for this evaluation only
  Boolean retval = client.getBooleanValue(flagKey, false, null,
          FlagEvaluationOptions.builder().hook(new ExampleHook()).build());
```